src/AccessibilityInsights.SharedUx/Behaviors/DropDownButtonBehavior.cs

```diff
@@ -90,7 +113,9 @@ protected override void OnDetaching()
         /// <param name="e"></param>
         void ContextMenu_Closed(object sender, RoutedEventArgs e)
         {
-            AssociatedObject.Background = origBackground;
```

Comment:
This was the cause of Problem 2 because it overrode the styling. Calling ClearValue restores the Background to its "default" state, which preserves the styling.